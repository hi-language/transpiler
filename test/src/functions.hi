// implicit return
add: (a, b) {
  a + b
}
_(add(5, 3))

// explicit return
greet: (name) {
  ^ "Hello, " + name
  _("this should not print")
}
_(greet("Master"))

// arrow function
multiply: (a, b) => a * b
_(multiply(2, 4))

// block with side effects and return
calculator: {
  add: (a, b) => a + b,
  subtract: (a, b) {
    _("Subtracting...")
    a - b
  }
}
_(calculator.subtract(10, 3))
