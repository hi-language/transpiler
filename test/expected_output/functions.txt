8
Hello, Master
8
Subtracting...
7
